src/components/offerStatus/StatusFormOnPage.js

```diff
@@ -0,0 +1,120 @@
+import { MaxLengthReason, OfferStatus, OfferType } from './constants';
+import { parseOfferId } from '../../utils/parseOfferId';
+import { parseOfferType } from '../../utils/parseOfferType';
+import { useChangeStatus as useChangeStatusPlace } from '../../hooks/api/places';
+import { useChangeStatus as useChangeStatusEvent } from '../../hooks/api/events';
+import { useTranslation } from 'react-i18next';
+import { ReasonAndTypeForm } from './ReasonAndTypeForm';
+
+const StatusFormOnPage = ({ offer, error }) => {
+  const { t, i18n } = useTranslation();
+  const router = useRouter();
+
+  const offerId = parseOfferId(offer['@id']);
+  const offerType = parseOfferType(offer['@context']);
+  const name =
+    offer?.name?.[i18n.language] ?? offer?.name?.[offer.mainLanguage];
+  const rawStatusType = offer?.status?.type;
+  const rawStatusReason = offer?.status?.reason;
+
+  const [type, setType] = useState('');
+  const [reason, setReason] = useState('');
+
+  useEffect(() => {
+    if (!rawStatusType) return;
+    setType(rawStatusType);
+  }, [rawStatusType]);
+
+  useEffect(() => {
+    const newReason = offer?.status?.reason?.[i18n.language];
+    if (!rawStatusReason || !newReason) return;
+    setReason(newReason);
+  }, [rawStatusReason]);
+
+  const useChangeStatus = useMemo(() => {
+    if (offerType === OfferType.PLACE) {
+      return useChangeStatusPlace;
+    }
+    return useChangeStatusEvent;
+  }, [offerType]);
+
+  const handleSuccessChangeStatus = () =>
+    router.push(`/${offerType}/${offerId}/preview`);
+
+  const { mutate: changeStatus, ...changeStatusMutation } = useChangeStatus({
+    onSuccess: handleSuccessChangeStatus,
+  });
+
+  return (
+    <Page>
+      <Page.Title>{t('offerStatus.title', { name })}</Page.Title>
+      <Page.Content spacing={5}>
+        {changeStatusMutation.status === QueryStatus.LOADING ? (
+          <Spinner marginTop={4} />
+        ) : error || changeStatusMutation.error ? (
+          <Alert variant={AlertVariants.WARNING}>
+            {error.message || changeStatusMutation.error?.message}
+          </Alert>
+        ) : (
+          [
+            <ReasonAndTypeForm
+              key="reason-and-type"
+              offerType={offerType}
+              statusType={type}
+              statusReason={reason}
+              onChangeStatusType={(e) => setType(e.target.value)}
+              onInputStatusReason={(e) => setReason(e.target.value)}
+            />,
+            <Inline key="actions" spacing={3}>
+              <Button
+                variant={ButtonVariants.SECONDARY}
+                onClick={() => router.push(`/offer/${offerId}/edit`)}
+              >
+                {t('offerStatus.actions.cancel')}
+              </Button>
+              <Button
+                variant={ButtonVariants.PRIMARY}
+                disabled={
+                  !offer ||
+                  reason.length === 0 ||
+                  reason.length > MaxLengthReason
+                }
```

Comment:
I think this is incorrect, the reason is always optional. So you should be able to set the status to `Unavailable` or `TemporarilyUnavailable` without a reason.

Reply:
Additionally, it should also be enabled when you pick "Available" (Gaat door/Open), because you need to be able to change the status back to that from Unavailable or TemporarilyUnavailable.

So I think a better would be something like the selected type not being the current type on the offer.

Reply:
in ticket: [III-3576](https://jira.uitdatabank.be/browse/III-3576)
this is written though `when a user chooses to update the status of his event to `available`, no reason can be added`

Reply:
Yes, that means the reason field should be disabled which is implemented correctly. My comment is about the save button.

Reply:
I meant that the save button should not be disabled just because the reason is empty.

Reply:
I think the behaviour now is as it should be then

Reply:
No because now the save button is disabled if you don't enter a reason? The reason should always be optional

Reply:
Ah you updated the disabled check, looks good now 👍